mealsharingdotnet/Program.cs

```diff
@@ -0,0 +1,167 @@
+{
+  return mealSharingService.ListMeals();
+});
+
+/*
+ <summary>
+ Endpoint for updating a blog posts by id.
+ <example>
+ postman input
+ <code>
+ http://localhost:5157/meals/1
+ </code>
+ </example>
+ </summary>
+*/
+app.MapPut("/meals/{id}", ([FromServices] IMealService mealSharingService, int id, Meal updateBody) =>
+{
+  mealSharingService.UpdateMeal(id, updateBody);
+});
+
+/*
+ <summary>
+ Endpoint for deletting a blog posts by id.
+ <example>
+ postman input
+ <code>
+ http://localhost:5157/meals/1
+ </code>
+ </example>
+ </summary>
+*/
+app.MapDelete("/meals/{id}", ([FromServices] IMealService mealSharingService, int id) =>
+{
+  mealSharingService.DeleteMeal(id);
+});
+
+app.Run();
+
+public interface IMealService
+{
+  List<Meal> ListMeals();
+  void AddMeal(Meal meal);
+  void DeleteMeal(int id);
+  void UpdateMeal(int id, Meal updateBody);
+}
+public class Meal
+{
+  public int ID { get; set; }
+
+  public string? Headline { get; set; }
+  public string? ImageURL { get; set; }
+  public string? BodyText { get; set; }
+  public string? Location { get; set; }
+  public int Price { get; set; }
+}
+public class FileMealService : IMealService
+{
+  // public List<Meal> listOfMeals = new List<Meal>();
+  public void AddMeal(Meal meal)
+  {
+    if (!File.Exists("meals.json"))
+    {
+      File.WriteAllText("meals.json", "[]");
+    }
+    var readJsonFile = File.ReadAllText(@"meals.json");
+    var meals = System.Text.Json.JsonSerializer.Deserialize<List<Meal>>(readJsonFile);
+    // listOfMeals.Add(meal);
+    var guid = Guid.NewGuid();
+    var ids = guid.ToString();
+    if (meals.Count == 0)
+    {
+      meal.ID = ids.Max(id => id);
+    }
+    else
+    {
+      foreach (var m in meals)
+      {
+        meal.ID = meals.Max(m => m.ID + 1);
+      }
+    }
+    meals.Add(meal);
+    var mealsJson = System.Text.Json.JsonSerializer.Serialize(meals);
+    File.WriteAllText("meals.json", mealsJson);
+  }
+  public List<Meal> ListMeals()
+  {
+    var readJsonFile = File.ReadAllText(@"meals.json");
+    var mealsFromJsonFile = System.Text.Json.JsonSerializer.Deserialize<List<Meal>>(readJsonFile);
+    // return listOfMeals;
+    return mealsFromJsonFile;
+  }
+  public void UpdateMeal(int id, Meal updateBody)
+  {
+    var readJsonFile = File.ReadAllText(@"meals.json");
+    var meals = System.Text.Json.JsonSerializer.Deserialize<List<Meal>>(readJsonFile);
+    var mealtoUpdate = meals.Find(findMealtoUpdate => findMealtoUpdate.ID == id);
+    mealtoUpdate.BodyText = updateBody.BodyText;
+    mealtoUpdate.Headline = updateBody.Headline;
+    mealtoUpdate.Price = updateBody.Price;
+    mealtoUpdate.ImageURL = updateBody.ImageURL;
+    var mealsJson = System.Text.Json.JsonSerializer.Serialize(meals);
+    File.WriteAllText("meals.json", mealsJson);
+  }
+  public void DeleteMeal(int id)
+  {
+    var readJsonFile = File.ReadAllText(@"meals.json");
+    var meals = System.Text.Json.JsonSerializer.Deserialize<List<Meal>>(readJsonFile);
```

Comment:
some improvement could be this:
you are using often this:
```
var readJsonFile = File.ReadAllText(@"meals.json");
var meals = System.Text.Json.JsonSerializer.Deserialize<List<Meal>>(readJsonFile);
```

you could extract it in a seprate function that would be private to the `FileMealService` .

so youd have something like 
```
Task<List<Meal>>ReadMealsFromFile(){
  var readJsonFile = File.ReadAllText(@"meals.json");
return await System.Text.Json.JsonSerializer.DeserializeAsync<List<Meal>>(readJsonFile);
}

```